lumicks/pylake/tests/test_imaging_confocal/test_kymo_slice_crop.py

```diff
@@ -0,0 +1,273 @@
+    np.testing.assert_allclose(kymo.get_image("red").data, ref.image[:, :, 0])
+
+    sliced = kymo[:]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :, 0])
+
+    # Anything will crop of the first frame
+    # need to make sure you're past the deadtime
+    # todo: for 2.0 define the start of the kymo as the start of the actual image, not deadtime
+    sliced = kymo[f"{deadtime_slice_offset}s":]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines - 1)
+    assert sliced.shape == (ref_pixels, ref_lines - 1, 3)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, 1:, 0])
+
+    sliced = kymo["0s":]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :, 0])
+
+    factor = 2
+    sliced = kymo["0s":f"{factor * line_time}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, factor)
+    assert sliced.shape == (ref_pixels, factor, 3)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :factor, 0])
+
+    sliced = kymo["0s":f"-{line_time * 0.6}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines - 1)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :-1, 0])
+
+    sliced = kymo["0s":f"-{2 * line_time}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines - 2)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, : (ref_lines - 2), 0])
+
+    # get a sliver of the next frame
+    # stop needs to be > halfway the deadtime between lines
+    sliced = kymo["0s":f"-{2 * line_time - deadtime_slice_offset}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines - 1)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :-1, 0])
+
+    # Two full frames
+    sliced = kymo["0s":f"{2 * line_time}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, 2)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :2, 0])
+
+    # Two full frames plus a bit
+    sliced = kymo["0s":f"{2 * scan_time + 2 * dead_time + deadtime_slice_offset}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, 3)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :3, 0])
+
+    sliced = kymo["3.5s":"14s"]
+    assert sliced.get_image("red").shape == (ref_pixels, 1)
+    assert sliced.shape == (ref_pixels, 1, 3)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, 1:2, 0])
+
+    sliced = kymo["0s":"85s"]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines)
+    assert sliced.shape == (ref_pixels, ref_lines, 3)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :, 0])
+
+    with pytest.raises(IndexError, match="Scalar indexing is not supported, only slicing"):
```

Comment:
Wonderful. I like having the actual error messages right here (saw this added in more places).